### ISSUE TITLE
How to retrieve the "s" command  so that my bezier curves are connected?

### ISSUE THREAD

**Issue by AlexandreMarquesDias** [memononen/nanosvg#231]
I'm using the simple example provided
```
for (shape = image->shapes; shape != NULL; shape = shape->next) {
	for (path = shape->paths; path != NULL; path = path->next) {
		for (i = 0; i < path->npts-1; i += 3) {
			float* p = &path->pts[i*2];
			drawCubicBez(p[0],p[1], p[2],p[3], p[4],p[5], p[6],p[7]);
		}
	}
}
```

And I created my own "drawCubicBez" which assigns a number of points per bezier curve and then connects a straight line between those points. The problem is, some paths contain a special SVG tag called "s" which indicates that two bezier lines need to be connected, but I don't know how to fetch this tag and because of this, my SVG images have non-continuous beziers.

Here's how a bunny SVG looks when I attempt to draw its curves : 

<img width="958" alt="problem" src="https://user-images.githubusercontent.com/57512424/200705738-c395b2c1-4736-4562-bff4-55662feb7eb2.PNG">

As you can see, encircled in red, there are "holes" in the middle of some of the bunny's curves because from one bezier to another, they are not connected. And they are not connected, because I don't know which curve should be connected together. Is there a way I could fix this? Knowing if a path contains the "s" command would help, I think.

**Comment by wcout** [comment]
Isn't this information in what is passed as the `closed` member of the `path` struct?

In `nanosvg/examples/example1.c` it is done like this:

````cpp
void drawPath(float* pts, int npts, char closed, float tol)
{
	int i;
	glBegin(GL_LINE_STRIP);
	glColor4ubv(lineColor);
	glVertex2f(pts[0], pts[1]);
	for (i = 0; i < npts-1; i += 3) {
		float* p = &pts[i*2];
		cubicBez(p[0],p[1], p[2],p[3], p[4],p[5], p[6],p[7], tol, 0);
	}
	if (closed) {
		glVertex2f(pts[0], pts[1]);
	}
	glEnd();
}

// ...
	for (shape = g_image->shapes; shape != NULL; shape = shape->next) {
		for (path = shape->paths; path != NULL; path = path->next) {
			drawPath(path->pts, path->npts, path->closed, px * 1.5f);
			drawControlPts(path->pts, path->npts);
		}
	}

// ...
	
````

**Comment by AlexandreMarquesDias** [comment]
> Isn't this information in what is passed as the `closed` member of the `path` struct?
> 
> In `nanosvg/examples/example1.c` it is done like this:
> 
> ```c++
> void drawPath(float* pts, int npts, char closed, float tol)
> {
> 	int i;
> 	glBegin(GL_LINE_STRIP);
> 	glColor4ubv(lineColor);
> 	glVertex2f(pts[0], pts[1]);
> 	for (i = 0; i < npts-1; i += 3) {
> 		float* p = &pts[i*2];
> 		cubicBez(p[0],p[1], p[2],p[3], p[4],p[5], p[6],p[7], tol, 0);
> 	}
> 	if (closed) {
> 		glVertex2f(pts[0], pts[1]);
> 	}
> 	glEnd();
> }
> 
> // ...
> 	for (shape = g_image->shapes; shape != NULL; shape = shape->next) {
> 		for (path = shape->paths; path != NULL; path = path->next) {
> 			drawPath(path->pts, path->npts, path->closed, px * 1.5f);
> 			drawControlPts(path->pts, path->npts);
> 		}
> 	}
> 
> // ...
> 	
> ```

Thank you very much for your answer. I figured out the bug I had, and it didn't have to do with anything related with a curve being closed or not.

In my "drawCubicBez" function, I had the following for loop, for adding vertices to a curve.

`for(double i=0;i<dotsPerCurve;i++)
  {`

This is erroneous, as I always missed out on the last dot on the curve I needed because of the i<dotsPerCurve condition.

The fixed loop looks like this : 
`for(double i=0;i<=dotsPerCurve;i++)
  {`

and the "holes" between curves disappear! 

Dumb little bug.
